ddp-workspace/projects/ddp-dsm-ui/src/app/shipping-search/shipping-search.component.ts

```diff
@@ -22,26 +24,40 @@ export class ShippingSearchComponent implements OnInit {
   kit: KitRequest[] = [];
   private currentPatchField: string | null;
   PECGS_RESEARCH = 'PECGS_RESEARCH';
+  disabled = true;
 
-  constructor(private dsmService: DSMService, private auth: Auth, private role: RoleService) {
+  constructor(private dsmService: DSMService, private auth: Auth, private role: RoleService, private compService: ComponentService) {
     if (!auth.authenticated()) {
       auth.sessionLogout();
     }
   }
 
   ngOnInit(): void {
+    if (sessionStorage.getItem(ComponentService.MENU_SELECTED_REALM) != null) {
+      this.realm = sessionStorage.getItem(ComponentService.MENU_SELECTED_REALM);
+    } else {
+      this.errorMessage = 'Please select a study';
+    }
     this.checkRight();
   }
 
   private checkRight(): void {
+    let allowedToSeeInformation = false;
     this.allowedRealms = [];
     let jsonData: any[];
     this.dsmService.getRealmsAllowed(Statics.SHIPPING).subscribe({
       next: data => {
         jsonData = data;
         jsonData.forEach((val) => {
-          this.allowedRealms.push(val);
+          if (sessionStorage.getItem(ComponentService.MENU_SELECTED_REALM) === val) {
+            allowedToSeeInformation = true;
+            this.allowedRealms.push(val);
+            this.disabled = false;
+          }
```

Comment:
Old code was not checking to see if selected realm is allowed to see Kit Search page.

Race condition issue is related to `getRealmsAllowed` request. If this request has not completed and user (playwright test is fast) has already filled out fields and clicked `Search Kit` button, error message `You are not allowed to see kit Information` will be shown.